### PR TITLE
[1.0.x] Porting serialization fix to release branch and configuring it to build the two libs that we need to service for UWP

### PR DESCRIPTION
**Pull request by SedarG**
@gkhanna79, @ericstj @yizhang82 , PTAL

### CODE REVIEW

**Comment by yizhang82** [comment]
@zhenlan @shmao FYI

**Comment by gkhanna79** [comment]
LGTM. I will suggest waiting on @ericstj or @weshaggard to look at this as well.

**Comment by karelz** [comment]
@gkhanna79 is the branch opened?

**Comment by gkhanna79** [comment]
Yes it is. I just sent the email.

File: src/packages.builds [hunk 1]
```diff
@@ -28,6 +28,14 @@
     </Project>
 
     <!-- add specific builds / pkgproj's here to include in servicing builds -->
+    <Project Include="System.Private.DataContractSerialization\pkg\System.Private.DataContractSerialization.builds">
+      <AdditionalProperties>$(AdditionalProperties)</AdditionalProperties>
+    </Project>
+
+    <Project Include="System.Linq.Expressions\pkg\System.Linq.Expressions.builds">
```
Comment (ericstj) [inline src/packages.builds]: You haven't actually made any changes to this library.

File: src/packages.builds [hunk 2]
```diff
@@ -28,6 +28,14 @@
     </Project>
 
     <!-- add specific builds / pkgproj's here to include in servicing builds -->
+    <Project Include="System.Private.DataContractSerialization\pkg\System.Private.DataContractSerialization.builds">
```
Comment (ericstj) [inline src/packages.builds]: You'll need to bump the version for this so that we get a new package version.  You may want to consider producing new versions of the facades on top of this, so that we have new *public* packages that folks can reference to get the fix.  Typically we don't ask folks to directly reference private packages.

Comment (SedarG) [inline src/packages.builds]: Thanks! I'll update the PR according to this shortly.

File: src/System.Runtime.Serialization.Json/pkg/System.Runtime.Serialization.Json.pkgproj [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <Version>4.0.2.0</Version>
+    <Version>4.0.3</Version>
```
Comment (gkhanna79) [inline src/System.Runtime.Serialization.Json/pkg/System.Runtime.Serialization.Json.pkgproj]: Should there be a ".0" after the "3"?

Comment (SedarG) [inline src/System.Runtime.Serialization.Json/pkg/System.Runtime.Serialization.Json.pkgproj]: Well, there are also instances where version is specified with 3 parts. and since that's more accurate I decided to keep it as is.

**Comment by gkhanna79** [comment]
Do we have the final review on this PR? If so, please send an updated email to shiproom with the changes that were done.

**Comment by SedarG** [comment]
The final review is not completed yet. I will update the shiproom as soon as I have it

File: src/System.Linq.Expressions/src/System.Linq.Expressions.builds [hunk 4]
```diff
@@ -6,19 +6,19 @@
...
     </Project>
-    <!-- NETCore50 must redistribute binaries due to shared library
     <Project Include="System.Linq.Expressions.csproj">
       <TargetGroup>netcore50</TargetGroup>
     </Project>
     <Project Include="System.Linq.Expressions.csproj">
       <TargetGroup>netcore50aot</TargetGroup>
-    </Project> -->
+    </Project> 
+    <!-- NETCore50 must redistribute binaries due to shared library
```
Comment (ericstj) [inline src/System.Linq.Expressions/src/System.Linq.Expressions.builds]: Delete this comment.  It's no longer relevant.

File: src/System.Linq.Expressions/pkg/System.Linq.Expressions.pkgproj [hunk 5]
```diff
@@ -1,5 +1,8 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <Version>4.3.1</Version>
```
Comment (ericstj) [inline src/System.Linq.Expressions/pkg/System.Linq.Expressions.pkgproj]: This is release/1.0.0, right?  In that case it should be 4.1.1.

File: src/System.Runtime.Serialization.Xml/ref/System.Runtime.Serialization.Xml.csproj [hunk 6]
```diff
@@ -2,7 +2,7 @@
 <Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
-    <AssemblyVersion>4.1.1.0</AssemblyVersion>
+    <AssemblyVersion>4.1.1.1</AssemblyVersion>
```
Comment (ericstj) [inline src/System.Runtime.Serialization.Xml/ref/System.Runtime.Serialization.Xml.csproj]: We shouldn't update the reference assembly in servicing.  Did something force you to do this only for this library?

Comment (SedarG) [inline src/System.Runtime.Serialization.Xml/ref/System.Runtime.Serialization.Xml.csproj]: I had "{ContractName} has mismatched compile ({referenceAssemblyVersion}) and runtime ({implementationVersion}) versions on {target}.".  I am taking back updating the version of the reference assembly and adding validation suppression.

File: src/System.Private.DataContractSerialization/src/System.Private.DataContractSerialization.csproj [hunk 7]
```diff
@@ -3,7 +3,7 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
     <AssemblyName>System.Private.DataContractSerialization</AssemblyName>
-    <AssemblyVersion>4.1.1.0</AssemblyVersion>
+    <AssemblyVersion>4.1.1.1</AssemblyVersion>
```
Comment (zhenlan) [inline src/System.Private.DataContractSerialization/src/System.Private.DataContractSerialization.csproj]: Just a NIT, I would prefer to give the assembly the same version of the package (ie. 4.1.2.0) for easy remember. It may make our life a little bit easier in the future in debugging given that there are so many different versions floating around.

File: src/System.Runtime.Serialization.Xml/pkg/ValidationSuppression.txt [hunk 8]
```diff
@@ -1,2 +1,3 @@
 // Remove once we can OOB this to desktop
 PermitPortableVersionMismatch=.NETFramework,Version=v4.6.1;.NETFramework,Version=v4.6
+PermitHigherCompatibleImplementationVersion=.NETCore,Version=v5.0;.NETStandard,Version=v1.6;.NETCoreApp,Version=v1.0
```
Comment (zhenlan) [inline src/System.Runtime.Serialization.Xml/pkg/ValidationSuppression.txt]: Just to understand, why is this change needed?

File: src/System.Runtime.Serialization.Xml/ref/System.Runtime.Serialization.Xml.csproj [hunk 9]
```diff
@@ -15,4 +15,4 @@
     <None Include="project.json" />
   </ItemGroup>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
-</Project>
\ No newline at end of file
+</Project>
```
Comment (zhenlan) [inline src/System.Runtime.Serialization.Xml/ref/System.Runtime.Serialization.Xml.csproj]: I cannot add a comment at line 5 above... does AssemblyVersion need to be updated as well?

Comment (SedarG) [inline src/System.Runtime.Serialization.Xml/ref/System.Runtime.Serialization.Xml.csproj]: Nope. See https://github.com/dotnet/corefx/pull/14526#discussion_r93324221

File: src/System.Runtime.Serialization.Xml/pkg/ValidationSuppression.txt (hunk 8 above)
Comment (SedarG) [inline src/System.Runtime.Serialization.Xml/pkg/ValidationSuppression.txt]: The implementation assembly version is not anymore matching the ref assembly. This causes a package validation error. This is, but, a known situation and we have means to suppress this error message when it's intentional. That's what this line is about; suppressing that error.

File: src/System.Private.DataContractSerialization/src/System.Private.DataContractSerialization.csproj (hunk 7 above)
Comment (SedarG) [inline src/System.Private.DataContractSerialization/src/System.Private.DataContractSerialization.csproj]: It used to be the case where the package version was matching the assembly version. however CoreFX has moved to another scheme where all the packages are being built with the same package version and the assembly version revs independently.

File: src/System.Runtime.Serialization.Xml/ref/System.Runtime.Serialization.Xml.csproj (hunk 9 above)
Comment (ericstj) [inline src/System.Runtime.Serialization.Xml/ref/System.Runtime.Serialization.Xml.csproj]: nit: you could leave this out of the commit if you want.

File: src/System.Runtime.Serialization.Xml/pkg/ValidationSuppression.txt (hunk 8 above)
Comment (zhenlan) [inline src/System.Runtime.Serialization.Xml/pkg/ValidationSuppression.txt]: Thanks for the explanation. Maybe worth adding a comment that will be helpful for future understanding.
